### PR TITLE
Decouple netowrk interface name

### LINKED ISSUE #52: MTU customizer unit fails if eth0 not available
**What happened**:

When creating a Shoot with an ubuntu image, the mtu-customizer.service (systemd) fails because it expects the existence of eth0 network interface.
The mtu customizer should the mtu of the primary network interface to a common value 1460. 

Tested on AWS with coreos and ubuntu.

**What you expected to happen**:
mtu-customizer service should have run successfully.
MTU should be 1460.

**How to reproduce it (as minimally and precisely as possible)**:

 - AWS Shoot with a machine type that is capable of [ENA](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/enhanced-networking-ena.html) 
like m5.large 
- Machine image ubuntu (used ubuntu in version 18.4.20190617)

Execute `systemctl status mtu-customizer.service` and observe that it is in failed status.
Execute `ifconfig | grp eth0` and see that eth0 is missing.

**Anything else we need to know?**:

Ubuntu defaults to using the ENA network interface that is not eth0.
**Environment**:

- Gardener version (if relevant):
- Extension version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by vpnachev**
**What this PR does / why we need it**:
Remove the hard dependency the network interface to be named `eth0`.
 
**Which issue(s) this PR fixes**:
Fixes #52

**Special notes for your reviewer**:

**Release note**:

```improvement operator
The `eth0`  interface name is no longer hard-coded, instead the non-virtual network interface names are now dynamically detected.
```

### CODE REVIEW

File: charts/gardener-extension-provider-aws/templates/daemonset-custom-mtu.yaml [hunk 1]
```diff
@@ -28,12 +28,21 @@ spec:
       initContainers:
       - name: mtu-customizer
         image: {{ include "image.alpine" . }}
-        command: ["ip", "link", "set", "dev", "eth0", "mtu", "1460"]
+        command:
+        - /mtu-customizer/mtu-customizer.sh
         securityContext:
           capabilities:
             add: ["NET_ADMIN", "SYS_TIME"]
+        volumeMounts:
```
Comment (rfranzke) [inline charts/gardener-extension-provider-aws/templates/daemonset-custom-mtu.yaml]: Can you please add an annotation checksum to the annotations of this pod sec?

File: pkg/webhook/controlplane/ensurer.go [hunk 2]
```diff
@@ -264,13 +264,18 @@ func (e *ensurer) EnsureAdditionalUnits(ctx context.Context, ectx genericmutator
 		command              = "start"
 		trueVar              = true
 		customMTUUnitContent = `[Unit]
-    Description=Apply a custom MTU to eth0
-    Requires=eth0.network
+Description=Apply a custom MTU to network interfaces
+After=network-online.target
```
Comment (rfranzke) [inline pkg/webhook/controlplane/ensurer.go]: Is `network-online.target` present everywhere?

Comment (zanetworker) [reply pkg/webhook/controlplane/ensurer.go]: isn't this somehow  tied to `NetworkManager` being on the system or not?

Comment (vpnachev) [reply pkg/webhook/controlplane/ensurer.go]: `network-online.target` is independent from implementation details like NetworkManger.

I've no strong opinion whether it should be `network.target` or `network-online.target`, see details about them here https://wiki.freedesktop.org/www/Software/systemd/NetworkTarget/#runningservicesafterthenetworkisup

File: charts/gardener-extension-provider-aws/templates/daemonset-custom-mtu.yaml (hunk 1 above)
Comment (vpnachev) [reply charts/gardener-extension-provider-aws/templates/daemonset-custom-mtu.yaml]: done

**Pull request merged**